### PR TITLE
(feat) O3-2255: Using REST endpoint to fetch address template and updated the tests for the address component

### PR DESCRIPTION
**Pull request by vasharma05**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
This PR updates the address component in the patient registration form to fetch the address configuration using the REST endpoint `/es/rest/v1/addresstemplate`. This improves the label provided for the different fields on the form and the translation is also handled on the backend itself.

Alongside, the test cases were updated with the new endpoint's results as mock and added new test suite for the address search component.

## Screenshots
image

## Related Issue
https://issues.openmrs.org/browse/O3-2255

## Other

*None.*

### CODE REVIEW

File: packages/esm-patient-registration-app/src/patient-registration/patient-registration.component.tsx [hunk 1]
```diff
@@ -65,16 +65,16 @@ export const PatientRegistration: React.FC<PatientRegistrationProps> = ({ savePa
...
-    if (addressTemplate) {
-      const addressTemplateXml = addressTemplate?.results[0].value;
-      if (!addressTemplateXml) {
-        return;
-      }
-      const { addressValidationSchema } = parseAddressTemplateXml(addressTemplateXml);
-      setValidationSchema((validationSchema) => validationSchema.concat(addressValidationSchema));
-    }
-  }, [inEditMode, addressTemplate, initialAddressFieldValues]);
+  // useEffect(() => {
```
Comment (ibacher) [inline packages/esm-patient-registration-app/src/patient-registration/patient-registration.component.tsx:68]: We can delete this right?

**Review by ibacher — COMMENTED** [review]
Apologies for semi-hijacking this to make some stricter types. Looks good apart from one bit of commented code left in.

**Comment by denniskigen** [comment]
Will the value for `Country` default to Cambodia? Would it be preferable to have dev3 default to an empty field instead if so?

**Comment by ibacher** [comment]
@denniskigen It looks like we're properly pulling defaults from the address template, so if someone puts a value in there, it will be the default, but we should be able to unset it.

File: packages/esm-patient-registration-app/src/patient-registration/patient-registration.component.tsx (hunk 1 above)
Comment (vasharma05) [reply packages/esm-patient-registration-app/src/patient-registration/patient-registration.component.tsx:68]: Yes, I missed it.

**Comment by vasharma05** [comment]
Thank you @ibacher for the strict typings!
I'll take care of the same from the next time.

Regarding the default @denniskigen, @gracepotma filed this [ticket](https://issues.openmrs.org/browse/O3-2281?filter=-1) to remove the defaults in the ref app.

File: packages/esm-patient-registration-app/src/patient-registration/patient-registration.component.tsx (hunk 1 above)
Comment (vasharma05) [reply packages/esm-patient-registration-app/src/patient-registration/patient-registration.component.tsx:68]: Hi @ibacher !
This brings to another question. 
Do we need the regex validation from the address template to work on the frontend?
Thanks!

Comment (ibacher) [reply packages/esm-patient-registration-app/src/patient-registration/patient-registration.component.tsx]: @vasharma05 Probably eventually, but it's not the highest priority.

**Review by ibacher — APPROVED** [review]

**Pull request merged**